cloudinit/net/ephemeral.py

```diff
@@ -281,28 +282,30 @@ def __init__(
         self,
         distro,
         iface=None,
-        connectivity_url_data: Optional[Dict[str, Any]] = None,
+        connectivity_urls_data: Optional[List[Dict[str, Any]]] = None,
         dhcp_log_func=None,
     ):
         self.iface = iface
         self._ephipv4: Optional[EphemeralIPv4Network] = None
         self.lease: Optional[Dict[str, Any]] = None
         self.dhcp_log_func = dhcp_log_func
-        self.connectivity_url_data = connectivity_url_data
+        self.connectivity_urls_data = connectivity_urls_data or []
         self.distro = distro
         self.interface_addrs_before_dhcp = netinfo.netdev_info()
 
     def __enter__(self):
         """Setup sandboxed dhcp context, unless connectivity_url can already be
         reached."""
-        if self.connectivity_url_data:
-            if net.has_url_connectivity(self.connectivity_url_data):
```

Comment:
This function is now dead code. Can you please remove it in a separate commit or a followup PR?